### PR TITLE
Gardening: Migrate test suite to GH issues p. 24

### PR DESCRIPTION
**Pull request by AnthonyLatsis**

### CODE REVIEW

File: test/AutoDiff/compiler_crashers_fixed/sr15566-vardecl-adjoint-values.swift [hunk 1]
```diff
@@ -4,7 +4,8 @@
 // Every so often this test crashes the linker on Linux
 // REQUIRES: rdar87254800
 
-// SR-15566: Differentiable functions with control flow yield an assertion failure: "location is a VarDecl, but SILDebugVariable is empty"
+// https://github.com/apple/swift/issues/57869
+// Differentiable functions with control flow yield an assertion failure
```
Comment (AnthonyLatsis) [inline test/AutoDiff/compiler_crashers_fixed/sr15566-vardecl-adjoint-values.swift:8]: The failure message is already pasted at the end of the file.

File: test/AutoDiff/compiler_crashers_fixed/issue-57869-vardecl-adjoint-values.swift [hunk 2]
```diff
@@ -4,7 +4,8 @@
 // Every so often this test crashes the linker on Linux
 // REQUIRES: rdar87254800
 
-// SR-15566: Differentiable functions with control flow yield an assertion failure: "location is a VarDecl, but SILDebugVariable is empty"
+// https://github.com/apple/swift/issues/57869
+// Differentiable functions with control flow yield an assertion failure
```
Comment (AnthonyLatsis) [inline test/AutoDiff/compiler_crashers_fixed/issue-57869-vardecl-adjoint-values.swift:8]: The failure message is already pasted at the end of the file.

**Review by xedin — APPROVED** [review]

**Comment by xedin** [comment]
@swift-ci please test

**Comment by AnthonyLatsis** [comment]
Note to self: fix typo in third commit message.

**Comment by AnthonyLatsis** [comment]
Validation testing results:

https://ci.swift.org/job/swift-PR-macos/4102
https://ci.swift.org/job/swift-PR-Linux/4013
https://ci-external.swift.org/job/swift-PR-windows/5160

Just a smoke test this time; I’ve only reworded a commit.

@swift-ci please smoke test

**Comment by AnthonyLatsis** [comment]
@swift-ci please clean smoke test macOS

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test macOS

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test macOS

**Pull request merged**